File: legacy/graphql-java-kickstart-annotations/src/main/java/com/example/demo/gql/dataloaders/AuthorsDataLoader.java
package com.example.demo.gql.dataloaders;

import com.example.demo.gql.types.Author;
import com.example.demo.service.AuthorService;
import lombok.RequiredArgsConstructor;
import org.dataloader.BatchLoader;

import java.util.List;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

@RequiredArgsConstructor
public class AuthorsDataLoader implements BatchLoader<String, Author> {
    final AuthorService authorService;

    @Override
    public CompletionStage<List<Author>> load(List<String> keys) {
        return CompletableFuture.supplyAsync(() ->
                this.authorService.getAuthorByIdIn(keys)
        );
    }
}

File: dgs-kotlin/src/main/kotlin/com/example/demo/gql/dataloaders/AuthorsDataLoader.kt
package com.example.demo.gql.dataloaders

import com.example.demo.gql.types.Author
import com.example.demo.service.AuthorService
import com.netflix.graphql.dgs.DgsDataLoader
import org.dataloader.BatchLoader
import java.util.UUID
import java.util.concurrent.CompletableFuture.completedFuture
import java.util.concurrent.CompletableFuture.supplyAsync
import java.util.concurrent.CompletionStage

@DgsDataLoader(name = "authorsLoader")
class AuthorsDataLoader(val authorService: AuthorService) : BatchLoader<UUID, Author> {
    override fun load(keys: List<UUID>): CompletionStage<List<Author>> = completedFuture(
        authorService.getAuthorByIdIn(keys)
    )
}

